src/server/pbsd_init.c

```diff
@@ -1829,6 +1829,12 @@ pbsd_init_reque(job *pjob, int change_state)
 		/* update the state, typically to some form of QUEUED */
 		svr_evaljobstate(pjob, &newstate, &newsubstate, 1);
 		(void)svr_setjobstate(pjob, newstate, newsubstate);
+		if (newstate == JOB_STATE_QUEUED)
+			/* count for this sate is increamented
+ 			 * in svr_setjobstate() and same will 
+ 			 * again increament by svr_enquejob()
+ 			 */
```

Comment:
Maybe reword your comment a little?  Something like: "The state count will be incremented in both svr_setjobstate() and svr_enquejo().  Decrement here so the counts do not get off."